src/Processors/Executors/ExecutorTasks.h

```diff
@@ -67,6 +72,8 @@ class ExecutorTasks
 
     void processAsyncTasks();
 
+    bool shouldSpawn() const { return idle_threads <= TOO_MANY_IDLE_THRESHOLD; }
```

Comment:
Maybe I miss something but I find the name "idle" in "idle_threads" weird. These describe spawned threads which are doing work and they are _not_ idling, right? Why not call the variable "spawned_threads" and the constant "TOO_MANY_SPAWNED_THREADS".

Reply:
`spawned_threads` should roughly equal to `use_threads`.

The "spawned" threads are not idle unless placed into `threads_queue`. This is why `idle_threads` is introduced, to dynamically tracked number of idle threads.